compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1433,7 +1433,7 @@ public void setForeachTypedBindingPatternType(BLangForeach foreachNode) {
                 varType = inferRecordFieldType(recordType);
                 break;
             case TypeTags.XML:
-                varType = BUnionType.create(null, symTable.xmlType, symTable.stringType);
+                varType = symTable.xmlType;
```

Comment:
I think there's a deviation here too.

The [spec](https://ballerina.io/spec/lang/master/#section_5.1.6) says for `xml<T>` the iteration value type is `T`.

So the following should work, right?

```ballerina
public function main() {
    xml<'xml:Element> el = xml `<foo>foo</foo><bar/>`;

    foreach 'xml:Element item in el {
        
    }
}
```

Reply:
Similarly, the `lang.xml:iterator()` also needs to change.

The following doesn't work either.

```ballerina
public function main() {
    xml<'xml:Element> el = xml `<foo>foo</foo><bar/>`;

    record {| 'xml:Element value; |}? nextResult = el.iterator().next();
}
```

Reply:
Maybe we can merge this PR first and then fix the `xml<T>` scenario.